File: Swift-C++-Testsuite/test/arithmetic-operators/Substraction/Swift/OverflowSub.swift
@main
struct MyMain {

    static func main() {
        // This call would crash with a normal add.
        // returns -9223372036854775808
        let a = subWithOverflow(x: Int.min)
    }

    // The & sign turns an subition to an substraction without overflow
    static func subWithOverflow(x: Int) -> Int {
        let tmp = x &- 1
        return tmp
    }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-1/OverflowSub.cpp

#include <iostream>

struct MyMain {

    static void main() {
        // This call would crash with a normal add.
        // returns -9223372036854775808
        int a = subWithOverflow(std::numeric_limits<int>::min());
    }

    // The & sign turns an subition to an substraction without overflow
    static int subWithOverflow(int x) {
        int tmp = x &- 1;
        return tmp;
    }
};

int main() {
    MyMain::main();

    return 0;
}